keras/applications/convnext.py

```diff
@@ -0,0 +1,549 @@
+  values in the [0-255] range.
+
+  Args:
+    include_top: Whether to include the fully-connected
+        layer at the top of the network. Defaults to True.
+    weights: One of `None` (random initialization),
+          `"imagenet"` (pre-training on ImageNet), or the path to the weights
+          file to be loaded. Defaults to `"imagenet"`.
+    input_tensor: Optional Keras tensor
+        (i.e. output of `layers.Input()`)
+        to use as image input for the model.
+    input_shape: Optional shape tuple, only to be specified
+        if `include_top` is False.
+        It should have exactly 3 inputs channels.
+    pooling: Optional pooling mode for feature extraction
+        when `include_top` is `False`. Defaults to None.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+    classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified. Defaults to 1000 (number of
+        ImageNet classes).
+    classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to `"softmax"`.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+class StochasticDepth(layers.Layer):
+    """Stochastic Depth module. It performs batch-wise dropping rather than
+      sample-wise. In libraries like `timm`, it's similar to `DropPath` layers
+      that drops residual paths sample-wise.
+
+    Reference:
+      - https://github.com.rwightman/pytorch-image-models
+
+    Args:
+      drop_path (float): Probability of dropping paths. Should be within [0, 1].
+    
+    Returns:
+      Tensor either with the residual path dropped or kept.
+
+    """
+
+    def __init__(self, drop_path, **kwargs):
+      super().__init__(**kwargs)
+      self.drop_path = drop_path
+
+    def call(self, x, training=False):
+      if training:
+        keep_prob = 1 - self.drop_path
+        shape = (tf.shape(x)[0],) + (1,) * (len(tf.shape(x)) - 1)
+        random_tensor = keep_prob + tf.random.uniform(shape, 0, 1)
+        random_tensor = tf.floor(random_tensor)
+        return (x / keep_prob) * random_tensor
+      return x
+
+    def get_config(self):
+      config = super().get_config()
+      config.update({"drop_path": self.drop_path})
+      return config
+
+
+class Block(tf.keras.Model):
```

Comment:
Reasons to implement the block like this:

* It has a trainable variable for layer scaling (gamma).
* If it was implemented as a layer the users won't have the flexibility to investigate the shapes and configs of the individual components of the blocks.

Reply:
Sounds good. But it needs to be serializable.

Reply:
Is there a way I could check that? 

I was able to serialize the weights of the instantiated model in `.h5` format.